### ISSUE TITLE
vmware-iso: Packer looks for IP addresses in the wrong place

### ISSUE THREAD

**Issue by carlosonunez** [hashicorp/packer#10408]
When filing a bug, please include the following headings if possible. Any
example text in this template can be deleted.

#### Overview of the Issue

Packer got stuck upon trying to SSH into a newly-created VM built from the `vmware-iso` builder. However, I was able to SSH into the machine manually with no issues. This confused me.

Upon setting `PACKER_LOG=1`, I discovered that `vmware-iso` fetches IP addresses from `/var/db/vmware/vmnet-dhcpd-vmnet8.leases`. Here is a sample from the logs:

```
2020/12/18 13:19:03 packer-builder-vmware-iso plugin: [DEBUG] Error getting SSH address: IP lookup failed: None of the found device(s) [vmnet8] has a DHCP lease for MAC 00:0c:29:f7:fd:f8
2020/12/18 13:19:08 packer-builder-vmware-iso plugin: Located networkmapper configuration file using Fusion6: /Library/Preferences/VMware Fusion/networking
2020/12/18 13:19:08 packer-builder-vmware-iso plugin: GuestIP discovered device matching nat: vmnet8
2020/12/18 13:19:08 packer-builder-vmware-iso plugin: Lookup up IP information...
2020/12/18 13:19:08 packer-builder-vmware-iso plugin: GuestAddress found MAC address in VMX: 00:0c:29:f7:fd:f8
2020/12/18 13:19:08 packer-builder-vmware-iso plugin: Trying DHCP leases path: /var/db/vmware/vmnet-dhcpd-vmnet8.leases
2020/12/18 13:19:08 packer-builder-vmware-iso plugin: Error while reading DHCP lease path file /var/db/vmware/vmnet-dhcpd-vmnet8.leases: open /var/db/vmware/vmnet-dhcpd-vmnet8.leases: no such file or directory
2020/12/18 13:19:08 packer-builder-vmware-iso plugin: IP lookup failed: None of the found device(s) [vmnet8] has a DHCP lease for MAC 00:0c:29:f7:fd:f8
```
This does not match how Vagrant retrieves this IP, which is through (correctly) interrogating `vmrun` for it with the `getGuestIPAddresses` command:

```
vmrun getGuestIPAddress /Users/ncarlos/path/to/packer/metadata/packer-vmware-iso.vmx
192.168.155.135
```
Please fix. Thanks in advance!

#### Reproduction Steps

1. Create a few test Linux VMs in VMware Fusion manually.
2. Create a new Packer manifest with the `vmware-iso` builder. Any options will do.
3. `packer build` that manifest. Set `PACKER_LOG=1` and `PACKER_LOG_PATH=/path/to/log`
4. Wait for VM to start. It will hang waiting for SSH.

### Packer version

```
$: packer version
Packer v1.6.5

Your version of Packer is out of date! The latest version
is 1.6.6. You can update by downloading from www.packer.io/downloads
```

### Simplified Packer Buildfile

```
{
  "builders": [
    {
      "type": "vmware-iso",
      "disk_size": 32000,
      "guest_os_type": "vmkernel6",
      "iso_url": "file:///path/to/packer/data/g.data/esxi.iso",
      "iso_checksum": "bd218f0f26510e690695116d8628e4af",
      "ssh_username": "root",
      "ssh_password": "password",
      "disk_adapter_type": "sata",
      "cpus": 2,
      "memory": 8092,
      "http_directory": "/path/to/packer/data/g.data/tkg-s",
      "network": "nat",
      "network_adapter_type": "vmxnet3",
      "boot_wait": "5s",
      "boot_command": [
        "<tab><wait>",
        " kernelopt=runweasel timeout=1 ks=http://{{ .HTTPIP }}:{{ .HTTPPort }}/ks.cfg<enter>"
      ],
      "vmx_remove_ethernet_interfaces": true,
      "vmx_data": {
        "featMask.vm.hv.capable": "Min:1",
        "vhv.enable": "true"
      },
      "ssh_timeout": "5m",
      "pause_before_connecting": "1m30s"
    }
  ],
  "post-processors": [
    {
      "type": "vagrant",
      "output": "/path/to/packer/data/g.data/tkg-s/esx.box"
    }
  ]
}
```

### Operating system and Environment details

MacOS 14.0 Big Sur

### Log Fragments and crash.log files

See description.

**Comment by chrisbalmer** [comment]
This looks like #10177 which appears to be fixed in 1.6.6.

**Comment by carlosonunez** [comment]
Thanks, Chris! Go ahead and dupe this if a fix is in the works.
